src/main/java/RacingGame.java

```diff
@@ -8,28 +11,53 @@ public RacingGame(RacingCarMovingRule racingCarMovingRule) {
         this.racingCarMovingRule = racingCarMovingRule;
     }
 
-    public RacingGame() {
-        this(new RandomRacingCarMovingRule());
-    }
-
-    public ArrayList<RacingStep> run(int numCars, int numSteps) {
+    public ArrayList<RacingStep> run(List<String> carNameList, int numSteps) {
+        ArrayList<RacingCar> racingCarList = buildRacingCarList(carNameList);
         ArrayList<RacingStep> racingStepList = new ArrayList<>();
-        ArrayList<RacingCar> racingCarList = new ArrayList<>();
-
-        for (int carId = 0; carId < numCars; carId++) {
-            racingCarList.add(new RacingCar(racingCarMovingRule, carId));
-        }
 
         racingStepList.add(new RacingStep(racingCarList));
 
         for (int step = 0; step < numSteps - 1; step++) {
-            racingCarList.stream()
-                    .forEach(racingCar -> racingCar.forward());
+            racingCarList.forEach(racingCar -> racingCar.forward());
 
             racingStepList.add(new RacingStep(racingCarList));
         }
 
         return racingStepList;
     }
 
+    private ArrayList<RacingCar> buildRacingCarList(List<String> carNameList) {
+        ArrayList<RacingCar> racingCarList = new ArrayList<>();
+
+        for (String carName : carNameList) {
+            racingCarList.add(new RacingCar(carName, racingCarMovingRule));
+        }
+
+        return racingCarList;
+    }
+
+    public static List<String> getWinnerNames(ArrayList<RacingStep> racingStepList) {
+        if (racingStepList.isEmpty()) {
+            return Collections.EMPTY_LIST;
+        }
+
+        RacingStep lastRacingStep = racingStepList.get(racingStepList.size() - 1);
+
+        int winnerPosition = getWinnerPosition(lastRacingStep);
+
+        List<String> winnerNames = lastRacingStep.getCarNameAndPositionList().stream()
+                .filter(nameAndPosition -> nameAndPosition.getPosition() == winnerPosition)
+                .map(nameAndPosition -> nameAndPosition.getName())
+                .collect(Collectors.toList());
+
+        return winnerNames;
+    }
+
+    private static int getWinnerPosition(RacingStep lastRacingStep) {
+        return lastRacingStep.getCarNameAndPositionList().stream()
+                    .mapToInt(nameAndPosition -> nameAndPosition.getPosition())
```

Comment:
```suggestion
                    .mapToInt(RacingStep.RacingCarNameAndPosition::getPosition)
```
메소드 레퍼런스를 활용할수 있을것 같아요 😄 
https://brunch.co.kr/@oemilk/204

Reply:
개인적으로 메서드 레퍼런스는 가독성이 안좋아서 잘 사용하지 않습니다. 받은 파라매터와 호출하는 인자의 관계가 보이지 않아서요.